cmake/libvpx.cmake

```diff
@@ -9,6 +9,51 @@ set(include_directories
     ${libvpx_loc}/source/config
 )
 
+# Convert the ASM files from the ARM Developer Suite 1.0.1 syntax to syntax
+# accepted by the platform's native assembler (e.g. GNU as from binutils).
+#
+function(target_ads2native_sources target_name src_loc)
+    find_package(Perl REQUIRED)
+
+    set(list ${ARGV})
+    list(REMOVE_AT list 0 1)
+
+    if (WIN32)
+        set(ads2native ${libvpx_loc}/source/libvpx/build/make/ads2armasm_ms.pl)
+    elseif (APPLE)
+        set(ads2native ${libvpx_loc}/source/libvpx/build/make/ads2gas_apple.pl)
+    else()
+        set(ads2native ${libvpx_loc}/source/libvpx/build/make/ads2gas.pl)
+    endif()
+
+    set(full_native_list "")
+    set(gen_dst ${CMAKE_CURRENT_BINARY_DIR}/gen)
+    set(include_directories ${include_directories} ${gen_dst} PARENT_SCOPE)
+
+    foreach (entry ${list})
+        set(ads_name ${src_loc}/${entry})
+        set(native_name ${gen_dst}/${entry}.S)
+
+        get_filename_component(native_name_loc ${native_name} DIRECTORY)
+        file(MAKE_DIRECTORY ${native_name_loc})
+
+        add_custom_command(
+        OUTPUT
+            ${native_name}
+        COMMAND
+            sh -c \"cat \'${ads_name}\' | ${PERL_EXECUTABLE} \'${ads2native}\' > \'${native_name}\'\"
```

Comment:
Well... looks weird 🤪

Reply:
Probably you should use [VERBATIM](https://cmake.org/cmake/help/latest/command/add_custom_command.html)

Reply:
Also, why cat is used instead of `<`?

Reply:
Is there a way to pipe standard input and standard output through an application in cmake?  That's the reason why I used the shell for it.

Reply:
I thought cmake passes the command to the shell, doesn't it?

Reply:
I'm not sure, I guess I can try it and see if it works

Reply:
It did indeed work, thanks!  With your help, I might even learn cmake :smile: